[BUG] Tests fail in integration in distributed mode after allowing nested types through in sort and shuffle

**Describe the bug**
``` =================================== FAILURES ===================================
05:33:42  _______________________ test_orderby_array[Array(Byte)] ________________________
05:33:42  
05:33:42  data_gen = Array(Byte)
05:33:42  
05:33:42      @pytest.mark.parametrize('data_gen', single_level_array_gens, ids=idfn)
05:33:42      def test_orderby_array(data_gen):
05:33:42  >       assert_gpu_and_cpu_are_equal_sql(
05:33:42              lambda spark : unary_op_df(spark, data_gen),
05:33:42              'array_table',
05:33:42              'select array_table.a, array_table.a[0] as first_val from array_table order by first_val',
05:33:42              conf=allow_negative_scale_of_decimal_conf)
05:33:42  
05:33:42  src/main/python/array_test.py:60: 
05:33:42  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
05:33:42  src/main/python/asserts.py:362: in assert_gpu_and_cpu_are_equal_sql
05:33:42      assert_gpu_and_cpu_are_equal_collect(do_it_all, conf)
05:33:42  src/main/python/asserts.py:336: in assert_gpu_and_cpu_are_equal_collect
05:33:42      _assert_gpu_and_cpu_are_equal(func, True, conf=conf)
05:33:42  src/main/python/asserts.py:328: in _assert_gpu_and_cpu_are_equal
05:33:42      assert_equal(from_cpu, from_gpu)
05:33:42  src/main/python/asserts.py:90: in assert_equal
05:33:42      _assert_equal(cpu, gpu, float_check=get_float_check(), path=[])
05:33:42  src/main/python/asserts.py:38: in _assert_equal
05:33:42      _assert_equal(cpu[index], gpu[index], float_check, path + [index])
05:33:42  src/main/python/asserts.py:31: in _assert_equal
05:33:42      _assert_equal(cpu[field], gpu[field], float_check, path + [field])
05:33:42  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
05:33:42  
05:33:42  cpu = [], gpu = [None, 5, -121, 76, -128, 42]
05:33:42  float_check = <function get_float_check.<locals>.<lambda> at 0x7f8e57a2df70>
05:33:42  path = [0, 'a']
05:33:42  
05:33:42      def _assert_equal(cpu, gpu, float_check, path):
05:33:42          t = type(cpu)
05:33:42          if (t is Row):
05:33:42              assert len(cpu) == len(gpu), "CPU and GPU row have different lengths at {}".format(path)
05:33:42              if hasattr(cpu, "__fields__") and hasattr(gpu, "__fields__"):
05:33:42                  for field in cpu.__fields__:
05:33:42                      _assert_equal(cpu[field], gpu[field], float_check, path + [field])
05:33:42              else:
05:33:42                  for index in range(len(cpu)):
05:33:42                      _assert_equal(cpu[index], gpu[index], float_check, path + [index])
05:33:42          elif (t is list):
05:33:42  >           assert len(cpu) == len(gpu), "CPU and GPU list have different lengths at {}".format(path)
05:33:42  E           AssertionError: CPU and GPU list have different lengths at [0, 'a']
05:33:42  
05:33:42  src/main/python/asserts.py:36: AssertionError
05:33:42  ----------------------------- Captured stdout call -----------------------------
05:33:42  ### CPU RUN ###
05:33:42  ### GPU RUN ###
05:33:42  ### COLLECT: GPU TOOK 0.8999466896057129 CPU TOOK 1.1249110698699951 ###
05:33:42  _______________________ test_orderby_array[Array(Short)] _______________________
05:33:42  
05:33:42  data_gen = Array(Short)
05:33:42  
05:33:42      @pytest.mark.parametrize('data_gen', single_level_array_gens, ids=idfn)
05:33:42      def test_orderby_array(data_gen):
05:33:42  >       assert_gpu_and_cpu_are_equal_sql(
05:33:42              lambda spark : unary_op_df(spark, data_gen),
05:33:42              'array_table',
05:33:42              'select array_table.a, array_table.a[0] as first_val from array_table order by first_val',
05:33:42              conf=allow_negative_scale_of_decimal_conf)
05:33:42  
05:33:42  src/main/python/array_test.py:60: 
05:33:42  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
05:33:42  src/main/python/asserts.py:362: in assert_gpu_and_cpu_are_equal_sql
05:33:42      assert_gpu_and_cpu_are_equal_collect(do_it_all, conf)
05:33:42  src/main/python/asserts.py:336: in assert_gpu_and_cpu_are_equal_collect
05:33:42      _assert_gpu_and_cpu_are_equal(func, True, conf=conf)
05:33:42  src/main/python/asserts.py:328: in _assert_gpu_and_cpu_are_equal
05:33:42      assert_equal(from_cpu, from_gpu)
05:33:42  src/main/python/asserts.py:90: in assert_equal
05:33:42      _assert_equal(cpu, gpu, float_check=get_float_check(), path=[])
05:33:42  src/main/python/asserts.py:38: in _assert_equal
05:33:42      _assert_equal(cpu[index], gpu[index], float_check, path + [index])
05:33:42  src/main/python/asserts.py:31: in _assert_equal
05:33:42      _assert_equal(cpu[field], gpu[field], float_check, path + [field])
05:33:42  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
05:33:42  
05:33:42  cpu = [None, -20249, -359, -23148], gpu = []
05:33:42  float_check = <function get_float_check.<locals>.<lambda> at 0x7f8e57393670>
05:33:42  path = [108, 'a']
05:33:42  
05:33:42      def _assert_equal(cpu, gpu, float_check, path):
05:33:42          t = type(cpu)
```

--- a/integration_tests/README.md
+++ b/integration_tests/README.md
@@ -141,7 +141,7 @@ You do need to have access to a compatible GPU with the needed CUDA drivers. The
 
 ### Runtime Environment
 
-`--runtime_env` is used to specify the environment you are running the tests in. Valid values are `databricks` and `emr`. This is generally used
+`--runtime_env` is used to specify the environment you are running the tests in. Valid values are `databricks`,`emr`,`dataproc` and `apache`. This is generally used
 when certain environments have different behavior, and the tests don't have a good way to auto-detect the environment yet.
 
 ### timezone

--- a/integration_tests/src/main/python/array_test.py
+++ b/integration_tests/src/main/python/array_test.py
@@ -15,6 +15,7 @@
 import pytest
 
 from asserts import assert_gpu_and_cpu_are_equal_collect, assert_gpu_and_cpu_are_equal_sql
+from conftest import is_dataproc_runtime
 from data_gen import *
 from pyspark.sql.types import *
 
@@ -54,7 +55,8 @@ def test_make_array(data_gen):
                 'array(a, b)',
                 'array(b, a, null, {}, {})'.format(s1, s2)))
 
-
+@pytest.mark.xfail(condition=is_dataproc_runtime(),
+                   reason='https://github.com/NVIDIA/spark-rapids/issues/1541')
 @pytest.mark.parametrize('data_gen', single_level_array_gens, ids=idfn)
 def test_orderby_array(data_gen):
     assert_gpu_and_cpu_are_equal_sql(
@@ -64,6 +66,8 @@ def test_orderby_array(data_gen):
         conf=allow_negative_scale_of_decimal_conf)
 
 
+@pytest.mark.xfail(condition=is_dataproc_runtime(),
+                   reason='https://github.com/NVIDIA/spark-rapids/issues/1541')
 @pytest.mark.parametrize('data_gen', [ArrayGen(ArrayGen(short_gen, max_length=10), max_length=10),
                                       ArrayGen(ArrayGen(string_gen, max_length=10), max_length=10)], ids=idfn)
 def test_orderby_array_of_arrays(data_gen):
@@ -73,6 +77,8 @@ def test_orderby_array_of_arrays(data_gen):
         'select array_table.a, array_table.a[0][0] as first_val from array_table order by first_val')
 
 
+@pytest.mark.xfail(condition=is_dataproc_runtime(),
+                   reason='https://github.com/NVIDIA/spark-rapids/issues/1541')
 @pytest.mark.parametrize('data_gen', [ArrayGen(StructGen([['child0', byte_gen],
                                                           ['child1', string_gen],
                                                           ['child2', float_gen]]))], ids=idfn)

--- a/integration_tests/src/main/python/conftest.py
+++ b/integration_tests/src/main/python/conftest.py
@@ -62,6 +62,9 @@ def is_databricks_runtime():
 def is_emr_runtime():
     return runtime_env() == "emr"
 
+def is_dataproc_runtime():
+    return runtime_env() == "dataproc"
+
 _limit = -1
 
 def get_limit():

--- a/integration_tests/src/main/python/struct_test.py
+++ b/integration_tests/src/main/python/struct_test.py
@@ -15,6 +15,7 @@
 import pytest
 
 from asserts import assert_gpu_and_cpu_are_equal_collect, assert_gpu_and_cpu_are_equal_sql
+from conftest import is_dataproc_runtime
 from data_gen import *
 from pyspark.sql.types import *
 
@@ -39,6 +40,8 @@ def test_make_struct(data_gen):
                 'named_struct("foo", b, "bar", 5, "end", a)'))
 
 
+@pytest.mark.xfail(condition=is_dataproc_runtime(),
+                   reason='https://github.com/NVIDIA/spark-rapids/issues/1541')
 @pytest.mark.parametrize('data_gen', [StructGen([["first", boolean_gen], ["second", byte_gen], ["third", float_gen]]),
                                       StructGen([["first", short_gen], ["second", int_gen], ["third", long_gen]]),
                                       StructGen([["first", long_gen], ["second", long_gen], ["third", long_gen]]),
@@ -50,6 +53,8 @@ def test_orderby_struct(data_gen):
         'select struct_table.a, struct_table.a.first as val from struct_table order by val')
 
 
+@pytest.mark.xfail(condition=is_dataproc_runtime(),
+                   reason='https://github.com/NVIDIA/spark-rapids/issues/1541')
 @pytest.mark.parametrize('data_gen', [StructGen([["first", string_gen], ["second", ArrayGen(string_gen)], ["third", ArrayGen(string_gen)]])], ids=idfn)
 def test_orderby_struct_2(data_gen):
     assert_gpu_and_cpu_are_equal_sql(

